internal/session/sessionwantsender.go

```diff
@@ -102,6 +105,8 @@ func newSessionWantSender(ctx context.Context, sid uint64, pm PeerManager, spm S
 
 	sws := sessionWantSender{
 		ctx:                      ctx,
```

Comment:
I'd:

1. Stop passing the context in from the outside.
2. Construct it and cancel it internally.

Reply:
Otherwise, shutdown won't completely shut down the session want sender.